tests/tests_geomstats/test_connection.py

```diff
@@ -118,6 +118,37 @@ def test_log_connection_metric(self, dim, point, base_point):
 
         self.assertAllClose(result, expected, atol=1e-6)
 
+    def test_geodesic_with_exp_connection(
+        self, dim, point, tangent_vec, n_times, n_steps, expected
+    ):
+        sphere = Hypersphere(dim)
+        connection = Connection(dim)
+        connection.christoffels = sphere.metric.christoffels
+        geo = connection.geodesic(
+            initial_point=point, initial_tangent_vec=tangent_vec, n_steps=n_steps
+        )
+        times = gs.linspace(0, 1, n_times)
+        geo = geo(times)
+        result = geo.shape
+
+        self.assertAllClose(result, expected, atol=1e-6)
```

Comment:
Can you please pass `atol` as an input argument? (This need will be more clear after merging #1517)

Reply:
As an input argument of the test function? Only for the two tests you mentioned?

Reply:
Thanks!

For now the idea is that new tests pass the `tol` when required because it will allow us to really check what tols will be required by using randomized data. (Later I'll have to review all the tests to see where tols are hard-coded)